Rubberduck.CodeAnalysis/QuickFixes/IntroduceLocalVariableQuickFix.cs

```diff
@@ -26,23 +27,58 @@ public override void Fix(IInspectionResult result)
         {
             var identifierContext = result.Target.Context;
             var enclosingStatmentContext = identifierContext.GetAncestor<VBAParser.BlockStmtContext>();
-            var instruction = IdentifierDeclarationText(result.Target.IdentifierName, EndOfStatementText(enclosingStatmentContext));
+            var instruction = IdentifierDeclarationText(result.Target.IdentifierName, EndOfStatementText(enclosingStatmentContext), FrontPadding(enclosingStatmentContext));
             _state.GetRewriter(result.Target).InsertBefore(enclosingStatmentContext.Start.TokenIndex, instruction);
         }
 
         private string EndOfStatementText(VBAParser.BlockStmtContext context)
         {
-            if (context.TryGetPrecedingContext<VBAParser.EndOfStatementContext>(out var endOfStmtContext))
+            if (!context.TryGetPrecedingContext<VBAParser.IndividualNonEOFEndOfStatementContext>(out var individualEndOfStmtContext))
             {
-                return endOfStmtContext.GetText();
+                return Environment.NewLine;
             }
 
-            return Environment.NewLine;
+            var endOfLine = individualEndOfStmtContext.endOfLine();
+
+            if (endOfLine?.commentOrAnnotation() == null)
+            {
+                return individualEndOfStmtContext.GetText();
+            }
+
+            //There is a comment inside the preceding endOfLine, which we do not want to duplicate.
+            var whitespaceContext = individualEndOfStmtContext.whiteSpace(0);
```

Comment:
shouldn't we use the last `whiteSpace` in the `individualEndOfStmtContext`

Reply:
If there is an `endOfLine`, there will be at most one.